android/cli/commands/_build.js

```diff
@@ -2818,90 +2825,45 @@ AndroidBuilder.prototype.copyResources = function copyResources(next) {
 			// nothing to encrypt, continue
 			return next();
 		}
-
-		// figure out which titanium prep to run
-		let titaniumPrep = 'titanium_prep';
-		if (process.platform === 'darwin') {
-			titaniumPrep += '.macos';
-		} else if (process.platform === 'win32') {
-			titaniumPrep += '.win32.exe';
-		} else if (process.platform === 'linux') {
-			titaniumPrep += '.linux' + (process.arch === 'x64' ? '64' : '32');
+		if (!cloak) {
+			this.logger.error('Could not encrypt assets!');
+			process.exit(1);
 		}
 
-		// encrypt the javascript
-		const titaniumPrepHook = this.cli.createHook('build.android.titaniumprep', this, function (exe, args, opts, done) {
-			this.logger.info(__('Encrypting JavaScript files: %s', (exe + ' "' + args.slice(1).join('" "') + '"').cyan));
-			appc.subprocess.run(exe, args, opts, function (code, out, err) {
-				if (code) {
-					return done({
-						code: code,
-						msg: err.trim()
-					});
-				}
-
-				// write the encrypted JS bytes to the generated Java file
-				const assetCryptDest = path.join(this.buildGenAppIdDir, 'AssetCryptImpl.java');
-				this.unmarkBuildDirFile(assetCryptDest);
-				fs.writeFileSync(
-					assetCryptDest,
-					ejs.render(fs.readFileSync(path.join(this.templatesDir, 'AssetCryptImpl.java')).toString(), {
+		this.logger.info('Encrypting javascript assets...');
+		Promise.all(
+			jsFilesToEncrypt.map(async file => {
+				const from = path.join(this.buildAssetsDir, file);
+				const to = path.join(this.buildBinAssetsResourcesDir, file);
+
+				this.logger.debug(__('Encrypting: %s', from.cyan));
+				const content = await promisify(fs.readFile)(from);
+				const output = await cloak.encrypt(content);
+				await promisify(fs.ensureDir)(path.dirname(to));
+				this.unmarkBuildDirFile(to);
+				return promisify(fs.writeFile)(to, output);
+			})
+		).then(async () => {
```

Comment:
<!--
  0 failure: 
  1 warning:  android/cli/comma...
  
  
  DangerID: danger-id-Danger;
  File: android/cli/commands/_build.js;
  Line: 2846;
-->

- :warning: android/cli/commands/_build.js line 2846 – Each then() should return a value or throw (promise/always-return)